test/extended/networking/multus.go

```diff
@@ -0,0 +1,118 @@
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/image"
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-network][Feature:Multus-CNI functionality]", func() {
+	var oc *exutil.CLI
+	var ns string // namespace
+
+	oc = exutil.NewCLI("multus-e2e")
+
+	f := oc.KubeFramework()
+	g.It("should ensure that multus-cni functions correctly", func() {
+		makeNamespaceScheduleToAllNodes(f)
+		ns = f.Namespace.Name
+		fmt.Println()
+		o.Expect(testMultus(f, oc, ns)).To(o.Succeed())
+	})
+})
+
+/*
+	Desc: 	Multus is already installed on origin. This test aims to verify the integrity of the installation.
+		  	It follows the steps listed in the multus quickstart guide.
+			(See: https://github.com/k8snetworkplumbingwg/multus-cni/blob/master/docs/quickstart.md#storing-a-configuration-as-a-custom-resource)
+	Overview: Creates a net-attach-def, creates a pod with an annotation that refers to it,
+			 and checks for newly created net1 interface.
+*/
+func testMultus(f *e2e.Framework, oc *exutil.CLI, ns string) error {
+	// Setup
+	var err error
+	podName := "multus-test-pod"
+
+	expectNoError(err)
+
+	g.By("attempting to use multus to create a pod with multiple network interfaces")
+
+	// 1. First create net-attach-def using bridgeCNI
+	nad_yaml := exutil.FixturePath("testdata", "net-attach-defs", "bridge-nad.yml")
+	g.By(fmt.Sprintf("calling oc create -f %s", nad_yaml))
+
+	err = oc.AsAdmin().Run("create").Args("-f", nad_yaml).Execute()
+	expectNoError(err)
+
+	oc.AsAdmin().Run("get").Args("network-attachment-definitions").Execute() // Print verification that it's come up
+
+	// 2. Then make a pod with the right annotation to use the net-attach-def
+	_ = launchTestMultusPod(f, podName)
+
+	// 3. Check that the pod is up
+	_, err = waitForTestMultusPod(f, ns, podName)
+	expectNoError(err)
+
+	// 4. Then need to inspect the pod and verify net1 interface exists.
+	output, err := oc.AsAdmin().Run("exec").Args("multus-test-pod", "--", "ip", "a", "show", "dev", "net1").Output()
+	expectNoError(err)
+	fmt.Println(output)
+	o.Expect(strings.Contains(output, "net1")).Should(o.Equal(true))
+
+	return nil
+}
+
+// launches a centos pod that sleeps
+func launchTestMultusPod(f *e2e.Framework, podName string) *kapiv1.Pod {
+	contName := fmt.Sprintf("%s-container", podName)
+	pod := &kapiv1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: kapiv1.PodSpec{
+			Containers: []kapiv1.Container{
+				{
+					Name:    contName,
+					Image:   image.LocationFor("docker.io/centos/httpd-24-centos8:latest"),
+					Command: []string{"/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"},
```

Comment:
We want to avoid adding unnecessary new images to the e2e suite. You can use `e2epod.CreateExecPodOrFail` (from `k8s.io/kubernetes/test/e2e/framework/pod`) to create a pod using the standard "agnhost" testing image (pass a callback as the `tweak` arg to that function so you can adjust the pod definition before it's created, eg to add the NAD annotation) and then use `f.ExecShellInPod` or the like to run `ip` in it